### DISCUSSION TITLE
nvim-telescope / fzf filtering based on project type?

### DISCUSSION THREAD

**Discussion by silverjam** [LazyVim/LazyVim#4405] [Q&A] [answered]
Recenlyt lazyvim seems to have started filtering based on project type.  I have a mixed Rust and Python project so when I'm on a Rust file the "Find Files" action (the default action for `<leader> <space>`) will only show Rust and related filetypes (like cargo's toml files).  The inverse happens if I'm on a Python file, only Python related files will be shown.  Is this a new behavior that was recently introduced (that's hopefully configurable?), or is something else going on?  Thanks!

**Comment by dpetka2001** [comment, marked as answer]
`<leader>fF`

**Reply by silverjam** [reply]
Thanks, it was indeed a problem with the default behavior of `<space><space>` which is bound to "Find files (Root dir)" -- in my mixed language project the "Root dir" would change based on where I was in the Git tree because (I'm guessing) it's looking for a project root like `Cargo.toml` or `pyproject.toml`.

For future reference.  I discovered this by just looking at the keymaps and doing some experiments.  Running `:nmap` will dump the keymap for normal mode.

The bindings for `<Space><Space>` and `<Space>ff` appear to be bound to the same thing:

![image](https://github.com/user-attachments/assets/580f4e5a-efe1-4b7c-b25e-b8d85e27d396)
![image](https://github.com/user-attachments/assets/51eac976-9aca-4fbd-b054-11938eb3e497)

The `<Space>fF` has better behavior for this mixed language project because it always searches the current working dir:

![image](https://github.com/user-attachments/assets/8f8abef9-321a-4180-beda-cc535083214a)

Finally, the solution for me was to just rebind `<Space><Space>` to look at git files, which is a better default for me:

```lua
vim.keymap.del("n", "<Space><Space>")
vim.keymap.set(
  "n",
  "<Space><Space>",
  ":Telescope git_files<CR>",
  { remap = false, silent = true, desc = "Find in git (*s)" }
)
```

**Comment by zodman** [comment]
trick: use `!filetype` on the telescope for ingore files
